src/core/src/sys/systems.rs

```diff
@@ -282,6 +282,12 @@ impl System {
     #[inline] pub fn size(&self) -> usize {self.particles.len()}
 
     /// Return the center-of-mass of a molecule
+    ///
+    /// # Warning
+    ///
+    /// This function does not check for the particles' positions'
+    /// nearest images. To use this function properly, make sure
+    /// that all particles of the molecule are adjacent.  
```

Comment:
> Is it in the case where we have some particles of the molecule on a side of the cell and some other particles on the other side?

Exactly. To avoid that, we can use `wrap_molecule`. It get's messy when an input file is used and parts of a molecule are already wrapped.